driver/utils/unicode_converter.h

```diff
@@ -0,0 +1,722 @@
+#pragma once
+
+#include "driver/platform/platform.h"
+#include "driver/utils/utils.h"
+
+#include <unicode/ustring.h>
+#include <unicode/ucnv.h>
+
+#include <string>
+#include <string_view>
+#include <vector>
+
+using ConverterPivotWideCharType = UChar;
+inline const std::string converter_pivot_wide_char_encoding = "UTF-16";
+
+template <typename CharType>
+inline auto make_raw_str(std::initializer_list<CharType> && list) {
+    return std::string{list.begin(), list.end()};
+}
+
+inline auto sameEncoding(const std::string & lhs, const std::string & rhs) {
+    return (ucnv_compareNames(lhs.c_str(), rhs.c_str()) == 0);
+}
+
+// Return the same string with signature prefix removed, if it matches the provided signature/BOM byte array.
+// Even though the signature is specified in bytes, it will be considered matching only if it matches up to some character boundary.
+template <typename CharType>
+inline auto consumeSignature(std::basic_string_view<CharType> str, const std::string_view & signature) {
```

Comment:
why not just `removeBOM` ?

Reply:
You mean some known `removeBOM` function or you suggest naming change?
"Signature" is more correct term than "BOM." "Consume" because, you usually "consume" from the beginning of a buffer/string/sequence, whereas "remove" could mean remove from anywhere.